src/api.rs

```diff
@@ -154,7 +154,8 @@ impl MiddleWare for EndpointMiddleware {
         let mut url_c = url.clone();
         let mut segs: Vec<&str> = url.path_segments().unwrap().collect();
         segs.insert(0, self.version.as_str());
-        url_c.path_segments_mut().unwrap().clear().extend(segs);
+        url_c.set_path("");
+        url_c = url_c.join(&segs.join("/")).unwrap();
```

Comment:
In my opinion this would be a good solution for that this prevents also the duplicated path encoding.
```suggestion
        url_c.set_path(format!("{}{}", self.version, url_c.path()).as_str());
```